gax/src/main/java/com/google/api/gax/retrying/TimedRetryAlgorithm.java

```diff
@@ -70,5 +72,13 @@
    * @param nextAttemptSettings attempt settings, which will be used for the next attempt, if
    *     accepted
    */
-  boolean accept(TimedAttemptSettings nextAttemptSettings);
+  boolean shouldRetry(TimedAttemptSettings nextAttemptSettings);
+
+  /**
+   * Returns {@code true} if the retrying process should be canceled, or {@code false} other.
+   *
+   * @param nextAttemptSettings attempt settings, which will be used for the next attempt, if
+   *     accepted
+   */
+  boolean shouldCancel(TimedAttemptSettings nextAttemptSettings);
```

Comment:
The retry algorithm is responsible for making a decision about what to do with the current retriable call. There are 3 possible outcomes for each call (basically a synonym of Future result in this context): 
- **succeded** (result accepted, `Future#get(`) will return result, `Future#isCancelled()` will be `false`); 
- **failed** (exception thorown on a last retry, `Future#get()` will throw `ExecutionException`, `Future#isCancelled()` will be `false`); 
- **canceled** (`Future#get()` will throw `CancellationException`, `Future#isCancelled()` will be `true`)

Just inverting shouldRetry does not allow to achieve this distinction between possible outcomes. Note how all possible combinations of two boolean values are already assigned in the following truth table, and there is no combination left to represent cancellation case:

shouldRetry | response (0) or exception(1) | retriable future
------------ | ------------- | -------------
0 | 0 | **success**
0 | 1 | **failed**, we gave up retrying
1 | 0 | **undefined**, we hope it succeeds next time, should be retried
1 | 1 | **undefined**, response returned, but not the one we want, should be retried
? | ? | **canceled**, for example when server returns response (not an error), saying that server operation was canceled, or it is sill an undefined case, but too much time has passed, so the polling should be canceled

Why do you think that a concept from the outer code is pushed here? I believe that cancel case was just missed in the original implementation, so adding it here.